File: Templates/typescript/basic-api-wrapper/setup-auth.ts
#!/usr/bin/env node

import * as readline from 'readline';
import { AuthManager } from './auth.js';
import { config } from './config.js';

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function question(prompt: string): Promise<string> {
  return new Promise((resolve) => {
    rl.question(prompt, resolve);
  });
}

async function setupAuth(): Promise<void> {
  console.log('🔐 MCP Server Authentication Setup');
  console.log('=====================================\n');
  
  const authManager = new AuthManager();
  
  try {
    console.log('This will help you set up secure authentication for your MCP server.');
    console.log('Your API token will be stored securely in your system keychain.\n');
    
    // Get API token
    const apiToken = await question('Enter your API token: ');
    if (!apiToken.trim()) {
      console.error('❌ API token is required');
      process.exit(1);
    }
    
    // Store token
    await authManager.setToken(apiToken.trim());
    
    console.log('\n✅ Authentication setup complete!');
    console.log('Your API token has been stored securely.');
    console.log('\nNext steps:');
    console.log('1. npm run build');
    console.log('2. npm start');
    
  } catch (error) {
    console.error('❌ Setup failed:', error);
    process.exit(1);
  } finally {
    rl.close();
  }
}

// Test token if --test flag is passed
async function testToken(): Promise<void> {
  console.log('🧪 Testing stored token...\n');
  
  const authManager = new AuthManager();
  
  try {
    const hasToken = await authManager.hasToken();
    if (hasToken) {
      const token = await authManager.getToken();
      console.log('✅ Token found:');
      console.log(`   API Token: ${token.substring(0, 8)}...`);
      console.log(`   Base URL: ${config.apiBaseUrl}`);
    } else {
      throw new Error('No token found');
    }
  } catch (error) {
    console.error('❌ No token found. Run setup first:', error);
    process.exit(1);
  }
}

// Main execution
async function main(): Promise<void> {
  const args = process.argv.slice(2);
  
  if (args.includes('--test')) {
    await testToken();
  } else {
    await setupAuth();
  }
}

main().catch(console.error); 

File: Templates/python/basic-api-wrapper/setup_auth.py
#!/usr/bin/env python3

import keyring
import getpass
from config import Config

def setup_authentication():
    """Setup secure authentication for the MCP server."""
    print("🔐 Basic API Wrapper MCP Server - Authentication Setup")
    print("=" * 50)
    
    print(f"\nThis will store your API token securely in the system keyring.")
    print(f"Service: {Config.KEYRING_SERVICE}")
    print(f"Username: {Config.KEYRING_USERNAME}")
    
    # Get API token from user
    api_token = getpass.getpass("\nEnter your API token: ").strip()
    
    if not api_token:
        print("❌ API token cannot be empty.")
        return False
    
    try:
        # Store in keyring
        keyring.set_password(Config.KEYRING_SERVICE, Config.KEYRING_USERNAME, api_token)
        print("✅ API token stored securely in keyring.")
        
        # Verify storage
        stored_token = keyring.get_password(Config.KEYRING_SERVICE, Config.KEYRING_USERNAME)
        if stored_token == api_token:
            print("✅ Token verification successful.")
            return True
        else:
            print("❌ Token verification failed.")
            return False
            
    except Exception as e:
        print(f"❌ Error storing token: {e}")
        return False

def remove_authentication():
    """Remove stored authentication."""
    try:
        keyring.delete_password(Config.KEYRING_SERVICE, Config.KEYRING_USERNAME)
        print("✅ Authentication removed successfully.")
    except keyring.errors.PasswordDeleteError:
        print("ℹ️ No authentication found to remove.")
    except Exception as e:
        print(f"❌ Error removing authentication: {e}")

def check_authentication():
    """Check if authentication is properly configured."""
    try:
        token = keyring.get_password(Config.KEYRING_SERVICE, Config.KEYRING_USERNAME)
        if token:
            print("✅ Authentication is configured.")
            return True
        else:
            print("❌ No authentication found.")
            return False
    except Exception as e:
        print(f"❌ Error checking authentication: {e}")
        return False

if __name__ == "__main__":
    import sys
    
    if len(sys.argv) > 1:
        command = sys.argv[1].lower()
        if command == "setup":
            setup_authentication()
        elif command == "remove":
            remove_authentication()
        elif command == "check":
            check_authentication()
        else:
            print("Usage: python setup_auth.py [setup|remove|check]")
    else:
        setup_authentication() 